### ISSUE TITLE
Unable to track users actions on map

### ISSUE THREAD

**Issue by jerdog** [hypertrack/example-android#2]
I forked the project, added my publishable key and then ran the project in the emulator - it boots, runs, I can "Assign an action" but when I select "Track users actions on map" I get a FC and the following appears in the Android Monitor

```
04-03 16:41:52.946 2640-2640/com.hypertrack.example_android D/AndroidRuntime: Shutting down VM                                                                
--------- beginning of crash
04-03 16:41:52.947 2640-2640/com.hypertrack.example_android E/AndroidRuntime: FATAL EXCEPTION: main
Process: com.hypertrack.example_android, PID: 2640
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.hypertrack.example_android/com.hypertrack.example_android.UserMapActivity}: android.view.InflateException: Binary XML file line #9: Binary XML file line #34: Binary XML file line #34: Error inflating class fragment
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
    at android.app.ActivityThread.-wrap12(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6119)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
 Caused by: android.view.InflateException: Binary XML file line #9: Binary XML file line #34: Binary XML file line #34: Error inflating class fragment
 Caused by: android.view.InflateException: Binary XML file line #34: Binary XML file line #34: Error inflating class fragment
 Caused by: android.view.InflateException: Binary XML file line #34: Error inflating class fragment
 Caused by: java.lang.RuntimeException: API key not found.  Check that <meta-data android:name="com.google.android.geo.API_KEY" android:value="your API key"/> is in the <application> element of AndroidManifest.xml
    at com.google.maps.api.android.lib6.drd.r.b(:com.google.android.gms.DynamiteModulesB:115)
    at com.google.maps.api.android.lib6.auth.e.a(:com.google.android.gms.DynamiteModulesB:155)
    at com.google.android.gms.maps.internal.CreatorImpl.b(:com.google.android.gms.DynamiteModulesB:6074)
    at com.google.android.gms.maps.internal.CreatorImpl.newMapFragmentDelegate(:com.google.android.gms.DynamiteModulesB:72)
    at com.google.android.gms.maps.internal.i.onTransact(:com.google.android.gms.DynamiteModulesB:62)
    at android.os.Binder.transact(Binder.java:499)
    at uqw.newMapFragmentDelegate(:com.google.android.gms:179)
    at com.google.android.gms.maps.internal.CreatorImpl.newMapFragmentDelegate(:com.google.android.gms:99)
    at uqv.onTransact(:com.google.android.gms:62)
    at android.os.Binder.transact(Binder.java:499)
    at com.google.android.gms.maps.internal.zzc$zza$zza.zzI(Unknown Source)
    at com.google.android.gms.maps.SupportMapFragment$zzb.zzJy(Unknown Source)
    at com.google.android.gms.maps.SupportMapFragment$zzb.zza(Unknown Source)
    at com.google.android.gms.dynamic.zza.zza(Unknown Source)
    at com.google.android.gms.dynamic.zza.onInflate(Unknown Source)
    at com.google.android.gms.maps.SupportMapFragment.onInflate(Unknown Source)
    at android.support.v4.app.Fragment.onInflate(Fragment.java:1184)
    at android.support.v4.app.FragmentManagerImpl.onCreateView(FragmentManager.java:3439)
    at android.support.v4.view.LayoutInflaterCompatHC$FactoryWrapperHC.onCreateView(LayoutInflaterCompatHC.java:47)
    at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:189)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:769)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
    at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:861)
    at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:861)
    at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
    at com.hypertrack.lib.HyperTrackMapFragment.onCreateView(HyperTrackMapFragment.java:592)
04-03 16:41:52.948 2640-2640/com.hypertrack.example_android E/AndroidRuntime:     at android.support.v4.app.Fragment.performCreateView(Fragment.java:2192)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1255)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1472)
    at android.support.v4.app.FragmentManagerImpl.addFragment(FragmentManager.java:1691)
    at android.support.v4.app.FragmentManagerImpl.onCreateView(FragmentManager.java:3440)
    at android.support.v4.app.FragmentController.onCreateView(FragmentController.java:120)
    at android.support.v4.app.FragmentActivity.dispatchFragmentsOnCreateView(FragmentActivity.java:378)
    at android.support.v4.app.BaseFragmentActivityHoneycomb.onCreateView(BaseFragmentActivityHoneycomb.java:33)
    at android.support.v4.app.FragmentActivity.onCreateView(FragmentActivity.java:79)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:777)
    at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
    at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
    at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
    at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:292)
    at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
    at com.hypertrack.example_android.UserMapActivity.onCreate(UserMapActivity.java:27)
    at android.app.Activity.performCreate(Activity.java:6679)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
    at android.app.ActivityThread.-wrap12(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6119)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

The error looks to be due to missing a Google Maps geo key? No instructions exist around this if that is the case

EDIT: the code for my project was forked to [Gitlab](https://gitlab.com/jerdog/ht-example-android)

**Comment by piyushgupta27** [comment]
Th crash happened because of Google Maps API Key missing in the AndroidManifest.xml file. The stacktrace shows the error. I'll add a compilation failure here so that it doesn't end up in a crash.

``` Caused by: java.lang.RuntimeException: API key not found.  Check that <meta-data android:name="com.google.android.geo.API_KEY" android:value="your API key"/> is in the <application> element of AndroidManifest.xml```

**Comment by jerdog** [comment]
@piyushgupta27 so you need to have an API_KEY to do this? If so we need to adjust the documentation to reflect that and point them in that direction

**Comment by piyushgupta27** [comment]
This API_KEY is to enable google maps views in app. The example app doesnt have a docs section yet. So i'll relevant info and links in the comments above the corresponding code.

**Comment by jerdog** [comment]
Right - I was referring to the README.md file.

**Comment by jerdog** [comment]
resolved with f4efa31
